tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,217 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run open-vm-tools testing against VMware ESXi
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_autotest::esxi_utils;
+use Time::Local;
+
+our $VM_POWERED_ON  = 'Powered on';
+our $VM_POWERED_OFF = 'Powered off';
+
+sub run {
+    my $vm_name = console('svirt')->name;
+    my $vm_id   = esxi_vm_get_vmid($vm_name);
+    my $vm_ip   = esxi_vm_public_ip($vm_id);
+    chomp($vm_ip);
+
+    record_info('Guest Name', $vm_name);
+    record_info('Guest ID',   $vm_id);
+    record_info('Guest IP',   $vm_ip);
+
+    die "The variable \$vm_id or \$vm_ip cannot be empty." if ($vm_id eq "" || $vm_ip eq "");
+
+    assert_script_run('zypper -n in open-vm-tools') if (script_run('rpmquery open-vm-tools'));
+    assert_script_run("/usr/bin/vmware-checkvm | grep 'good'");
+    systemctl('status vmtoolsd');
+    systemctl('status vgauthd');
+    systemctl('restart vmtoolsd');
+    assert_script_run("systemctl status vmtoolsd | grep 'Started open-vm-tools'");
+    assert_script_run("/usr/bin/vmtoolsd -v | grep 'VMware Tools daemon, version'");
+    assert_script_run('vmware-toolbox-cmd logging level set vmtoolsd debug');
+    assert_script_run("vmware-toolbox-cmd logging level get vmtoolsd | grep 'vmtoolsd.level = debug'");
+    assert_script_run('vmware-toolbox-cmd logging level set vmtoolsd message');
+    assert_script_run("vmware-toolbox-cmd logging level get vmtoolsd | grep 'vmtoolsd.level = message'");
```

Comment:
How about moving above block into some functions, similar with below sub-tests like do_xx_tests?

Reply:
Agree. These commands looks more like some checks, so might be better call it do_sanity_checks().

Reply:
OK, already added empty lines.